trio/_core/_tests/test_instrumentation.py

```diff
@@ -193,26 +199,27 @@ async def main():
     # the TaskRecorder kept going throughout, even though the BrokenInstrument
     # was disabled
     assert ("after", main_task) in r.record
-    assert ("after_run",) in r.record
+    assert ("after_run", None) in r.record
     # And we got a log message
+    assert caplog.records[0].exc_info is not None
     exc_type, exc_value, exc_traceback = caplog.records[0].exc_info
     assert exc_type is ValueError
     assert str(exc_value) == "oops"
     assert "Instrument has been disabled" in caplog.records[0].message
 
 
-def test_instruments_monkeypatch():
+def test_instruments_monkeypatch() -> None:
     class NullInstrument(_abc.Instrument):
         pass
 
     instrument = NullInstrument()
 
-    async def main():
-        record = []
+    async def main() -> None:
+        record: list[Task] = []
 
         # Changing the set of hooks implemented by an instrument after
         # it's installed doesn't make them start being called right away
-        instrument.before_task_step = record.append
+        instrument.before_task_step = record.append  # type: ignore[assignment]
```

Comment:
I'm not sure why this is type erroring?

Reply:
```
trio/_core/_tests/test_instrumentation.py:222: error: Cannot assign to a method  [method-assign]
trio/_core/_tests/test_instrumentation.py:222: error: Incompatible types in assignment (expression has type "Callable[[Task], None]", variable has type "Callable[[Arg(Task, 'task')], None]")  [assignment]
```

If I edit `_abc.Instrument.before_task_step` to make `task` positional-only then I only get the first error. Seems like `ignore[assignment]` covers both, but made it extra explicit by ignoring both, and added comment that explains the errors.